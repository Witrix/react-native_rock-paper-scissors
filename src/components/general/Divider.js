import React, { Component } from 'react';
import { View }             from 'react-native';

const DEFAULT_SIZE = 1;

export default class Divider extends Component {
    constructor(props) {
        super(props);
        let size = DEFAULT_SIZE;
        if (props.size && typeof props.size === 'number' && props.size >= 0) size = props.size;
        this.state = {
            size: size,
        }
    }

    render() {
        return (
            <View style={{height: this.state.size || DEFAULT_SIZE, backgroundColor: 'grey', marginHorizontal: 20, marginVertical: 10}}/>
        );
    }
}
