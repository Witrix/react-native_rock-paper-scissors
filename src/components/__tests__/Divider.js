import 'react-native';
import React      from 'react';
import Divider    from '../general/Divider';
import renderer   from 'react-test-renderer';

it('Render correctly: No props', () => {
  const tree = renderer.create(
    <Divider />
  );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Correct props', () => {
    const tree = renderer.create(
        <Divider size={20}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Size null', () => {
    const tree = renderer.create(
        <Divider size={null}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Size invalid', () => {
    const tree = renderer.create(
        <Divider size={-1}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Size invalid', () => {
    const tree = renderer.create(
        <Divider size={"abc"}/>
    );
    expect(tree).toMatchSnapshot();
});

