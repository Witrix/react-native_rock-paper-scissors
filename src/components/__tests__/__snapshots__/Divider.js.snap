// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly: Correct props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "grey",
      "height": 20,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
/>
`;

exports[`Render correctly: No props 1`] = `
<View
  style={
    Object {
      "backgroundColor": "grey",
      "height": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
/>
`;

exports[`Render correctly: Size invalid 1`] = `
<View
  style={
    Object {
      "backgroundColor": "grey",
      "height": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
/>
`;

exports[`Render correctly: Size invalid 2`] = `
<View
  style={
    Object {
      "backgroundColor": "grey",
      "height": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
/>
`;

exports[`Render correctly: Size null 1`] = `
<View
  style={
    Object {
      "backgroundColor": "grey",
      "height": 1,
      "marginHorizontal": 20,
      "marginVertical": 10,
    }
  }
/>
`;
